### PR TITLE
[PROPOSAL] add proposal for odo component configuration

### PR DESCRIPTION
**Pull request by syamgk**
proposing a way for users to deal with environment variables
and configuration files for the applications
deployed on openshift cluster using odo.
fix issue #675

### CODE REVIEW

File: docs/proposals/odo-config-file.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#### Abstracting config map and Secrets from the user 
+
```
[the remark this answers is not in the document]
Comment (syamgk) [reply docs/proposals/odo-config-file.md]: added motivation section and ready for review round 2 :)

File: docs/proposals/odo-config-file.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#### Abstracting config map and secrets from the user 
+
+By default values will come from configmaps or secrets, 
+the name of configmap/secret keys will be same as environment variable names.
+configmap name should be unique for each component, 
+so using a combination of appname + component name. eg: <app_name>-<component_name>
+
+*Problem with this way of implementing:*
+If the user goes into configmap using oc or webUI and modify key names then it becomes messy.
```
Comment (anmolbabu) [inline docs/proposals/odo-config-file.md]: Did we find a solution to this?
We are like any ideal cli tool, as much stateless as possible right?
So, any operation on configmap via Odo is expected to fetch the configmap from openshift api and then permit/error out the operation seeking the then correct values from the user right?
It is very much possible that I do not have the full context, if so, please help me understand what the problem is

Comment (syamgk) [reply docs/proposals/odo-config-file.md]: Here the logic is to keep the `env variable names` same as `config map/secret keys names` and nothing being stored in local machine. So If the user adds a new configmap key or a new environment variable out of our scope then we wont process them as config instead will show a warning.

Comment (anmolbabu) [reply docs/proposals/odo-config-file.md]: Ok

**Pull request closed without merging**